CRM/MembershipExtras/Page/EditContributionRecurLineItems.php

```diff
@@ -105,11 +105,38 @@ private function isAllowedMembershipType($membershipType, $currentLineItems, $pe
       if ($membershipType['member_of_contact_id'] == $lineItemMembershipType['member_of_contact_id']) {
         return FALSE;
       }
+
+      $membershipForContact = $this->membershipTypeExistsForContact(
+        $lineItem['price_field_id'],
+        $membershipType['id']
+      );
+      if ($membershipForContact) {
+        return FALSE;
+      }
     }
 
     return TRUE;
   }
 
+  /**
+   * Searches for membership type in contact's active memberships.
+   *
+   * @param $priceFieldID
+   * @param $membershipTypeID
+   *
+   * @return bool
+   */
+  private function membershipTypeExistsForContact($priceFieldID, $membershipTypeID) {
+    $membershipResult = civicrm_api3('PriceFieldValue', 'getcount', [
```

Comment:
this API call does not really filter the membership type of a contact, both price_field_id and membership_type_id fields that you are using for filtration are generic fields and may return records for other contacts.